Sources/Testing/Events/Recorder/Event.HumanReadableOutputRecorder.swift

```diff
@@ -0,0 +1,378 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension Event {
+  /// A type which handles ``Event`` instances and outputs representations of
+  /// them as human-readable messages.
+  ///
+  /// This type can be used compositionally to produce output in other
+  /// human-readable formats such as rich text or HTML.
+  ///
+  /// The format of the output is not meant to be machine-readable and is
+  /// subject to change. For machine-readable output, use ``JUnitXMLRecorder``.
+  public struct HumanReadableOutputRecorder: Sendable {
+    /// A type describing a human-readable message produced by an instance of
+    /// ``Event/HumanReadableOutputRecorder``.
+    public struct Message: Sendable {
+      /// The symbol associated with this message, if any.
+      var symbol: Symbol?
+
+      /// The human-readable message.
+      var stringValue: String
+    }
+
+    /// A type that contains mutable context for
+    /// ``Event/ConsoleOutputRecorder``.
+    private struct _Context {
+      /// The instant at which the run started.
+      var runStartInstant: Test.Clock.Instant?
+
+      /// The number of tests started or skipped during the run.
+      ///
+      /// This value does not include test suites.
+      var testCount = 0
+
+      /// The number of test suites started or skipped during the run.
+      var suiteCount = 0
+
+      /// A type describing data tracked on a per-test basis.
+      struct TestData {
+        /// The instant at which the test started.
+        var startInstant: Test.Clock.Instant
+
+        /// The number of issues recorded for the test.
+        var issueCount = 0
+
+        /// The number of known issues recorded for the test.
+        var knownIssueCount = 0
+      }
+
+      /// Data tracked on a per-test basis.
+      var testData = Graph<String, TestData?>()
+    }
+
+    /// This event recorder's mutable context about events it has received,
+    /// which may be used to inform how subsequent events are written.
+    @Locked private var _context = _Context()
+
+    /// Initialize a new human-readable event recorder.
+    ///
+    /// - Parameters:
```

Comment:
Do zero-param declarations actually need `- Parameters:` ?

Reply:
Copy-and-paste error. :)